src/renderer/components/data-settings/data-settings.js

```diff
@@ -1056,7 +1055,7 @@ export default defineComponent({
         const channel = await getLocalChannel(channelId)
 
         if (channel.alert) {
-          return undefined
+          return []
```

Comment:
I know that returning an empty array is inline with the other parts of this file, however I think we should change that in all places, because we don't check for the empty array anywhere.

e.g. on this line we check if the `author` property of the return value is `undefined`, so when we return an empty array we are checking if `[].author` is `undefined`, as that property never exists, the check succeeds. Checking an array for a property that never exists seems wrong to me, either we should be checking for an empty array, or change the return values to be something other than an empty array, so that the checks make more sense.
https://github.com/FreeTubeApp/FreeTube/blob/4a86fa1b718b2b01c0c5c1a35c5bb1820b0f4f76/src/renderer/components/data-settings/data-settings.js#L1112

Reply:
Separate PR? (Since I have no idea what do you mean

Reply:
Also more difficult to review with more changes, this is just to fix importing

Reply:
Doing it in a separate pull request sounds good to me. The current solution works, so technically the bug is fixed, it's just that the way it's done in the whole file, including this pull request, is bad and makes the code confusing. I suspect it's leftover from a previous refactoring.

Hopefully chunky understands what I mean, otherwise I can do it myself.